src/bindgen/cargo/cargo_metadata.rs

```diff
@@ -180,16 +180,51 @@ impl Eq for Dependency {}
 pub fn metadata(
     manifest_path: &Path,
     existing_metadata_file: Option<&Path>,
+    all_platforms: bool,
 ) -> Result<Metadata, Error> {
     let output;
     let metadata = match existing_metadata_file {
         Some(path) => Cow::Owned(std::fs::read_to_string(path)?),
         None => {
+            let mut target = None;
+
+            if !all_platforms {
+                // cargo metadata defaults to giving information for _all_ targets.
+                // We figure out the host platform through rustc and use that.
+                // We unfortunatelly cannot go through cargo, since cargo rustc _also_ builds.
+                // If `rustc` fails to run, we just fall back to not passing --filter-platforms.
+                //
+                // NOTE: We set the current directory in case of rustup shenanigans.
+                let rustc = Command::new("rustc")
```

Comment:
This should probably use `env::var("RUSTC").unwrap_or_else...` like `CARGO` below.